File: go/SingleNumber/SingleNumber.go
package main

import "fmt"

// Time -> 0(N) Space -> 0(1)
func singleNumber(nums []int) int {
	val := 0
	for _, num := range nums {
		val ^= num
	}
	return val
}

func main() {
	fmt.Println(singleNumber([]int{1, 1, 2, 3, 4, 3, 2}))
}

File: python/singleNumber.py
# Time -> 0(N) Space -> 0(1)
def singleNumber(nums: list[int]) -> int:
    single = 0

    for val in nums:
        single ^= val
    return single


print(singleNumber([1, 1, 2, 3, 4, 3, 2]))
